Perform Maven target dependency update in non-UI thread

ptziegler opened the pull request:
Depending on how many artifacts are updated, this might be a long-running operation and should therefore be run without blocking the application.

laeubi commented:
@ptziegler can you check with latest release? I Think that I should have fixed this in PDE so it performs updates in a separate thread. If not let me know I think this is better handled in the PDE editor than directly in the target location implementation.

ptziegler commented:
> @ptziegler can you check with latest release? I Think that I should have fixed this in PDE so it performs updates in a separate thread. If not let me know I think this is better handled in the PDE editor than directly in the target location implementation.

I've just tested it with org.eclipse.pde 3.13.3000.v20250228-0140 and org.eclipse.m2e.pde.target 2.1.1.20250223-0727 and the problem still exists. 

More specifically, the issue lies with the following button, which is used to updated all selected dependencies. Those updates are done in the UI thread and if, like in my case, 200 or so artifacts are updated at once, the IDE freezes for ~1-2 minutes.

![image](https://github.com/user-attachments/assets/461fd1cc-7fbc-4b1e-b6b1-048cd69c42f9)

Given that this is purely an interaction between the dialog and Maven Central, I'm not sure if this has anything to do with PDE editor itself.

laeubi commented:
> Given that this is purely an interaction between the dialog and Maven Central, I'm not sure if this has anything to do with PDE editor itself.

I see, I thought this is referencing to the update button in the target editor, so is there anything left here or can we merge this after conflicts are resolved?

ptziegler commented:
I've rebased everything and fixed the merge conflict. Unless there are any related test failures, that should now include everything. If anyone is interested, this is how it looks like:

![image](https://github.com/user-attachments/assets/e8a7d1ee-21e9-40b2-835a-9bf8e27c278d)
